qa/rpc-tests/wallet.py

```diff
@@ -317,6 +317,46 @@ def run_test (self):
         except JSONRPCException as e:
             assert("Invalid or non-wallet transaction id" not in e.error['message'])
 
+        self.sync_all()
+
+        # test multiple private key import, and watch only address import
+        bal = self.nodes[2].getbalance()
+        addrs = [ self.nodes[1].getnewaddress() for i in range(0,20)]
+        pks   = [ self.nodes[1].dumpprivkey(x) for x in addrs]
+        for a in addrs:
+            self.nodes[0].sendtoaddress(a, 1)
+        self.nodes[0].generate(1)
+        self.nodes[2].importprivatekeys(pks[0], pks[1])
+        time.sleep(1)
+        assert(bal + 2 == self.nodes[2].getbalance())
+        self.nodes[2].importprivatekeys("rescan", pks[2], pks[3])
+        time.sleep(1)
+        assert(bal + 4 == self.nodes[2].getbalance())
+        self.nodes[2].importprivatekeys("no-rescan", pks[4], pks[5])
+        time.sleep(1)
+        assert(bal + 4 == self.nodes[2].getbalance())  # since the recan didn't happen, there won't be a balance change
+        self.nodes[2].importaddresses("rescan") # force a rescan although we imported nothing
+        time.sleep(1)
```

Comment:
are time.sleep(1) really necessary.  If so, is there a better way to wait?  If you can't get around it fine, but using them for syncronization often ends up with test failures and more debugging at some future date.

Reply:
The RPC calls spawn a rescan thread rather than doing it inline.  So I need to wait until the rescan is complete.  There is no way to check for that AFAIK.  I didn't want to add a new RPC.  Maybe I can add a "status" field to getinfo.